docs/adr/0000-logging-warnings-using-the-python-warnings-or-logging-module.md

```diff
@@ -24,9 +24,13 @@ Python has two different ways to report warnings to the user. Warnings can be lo
 ## Decision Outcome
 
 - The `warnings` module will be configured as such that it ignores warnings logged from non-inmanta python modules. This way warnings from third-party libraries are ignored.
-- All warnings, for which we expect an action from the end-user, should be logged using the `warnings.warn()` method. An example is a log message that announces a feature deprecation. This user can be a user of the CLI, the API, a module or an extension developer.
+- All warnings, for which we expect an action from the end-user, should be logged using `warnings.warn()` method, and inherit from the InmantaWarning type. An example is a log message that announces a feature deprecation. This user can be a user of the CLI, the API, a module or an extension developer. Those warning won't contain the python trace.
 - All other warnings should be logged using the `LOGGER.warning()` method. These warnings usually indicate that something went wong at runtime. For example, an agent that hits the rate limiter.
 
+## Rationale
+
+- We don't want to display python traces to end users as they contain no useful information and are clogging the logs. To prevent this we use the InmantaWarning which use a different formatting rule.
```

Comment:
One of the main reasons why we want to use `InmantaWarnings` is because the location were we log the warning is not always the location that the warning is about. For example, when deprecated syntax is used in the inmanta model, we log a warning in core and the default warning formatter will include the line in the core in the logs, while the line in the model is important. By using InmantaWarnings, we can manually include the correct filename and line number from the model in the warning message. That rationale is still missing here. Can you add it?